### ISSUE TITLE
ASP.NET Core 2.1 UseHttpsRedirection on macOS/kestrel not working

### ISSUE THREAD

**Issue by anorborg** [aspnet/Docs#6914]
I was working on a fairly empty project (made from the `dotnet new webapi` template. Attempted to use `app.UseHttpsRedirection`. While the page loads with https, if I use an `http://localhost:5001` directly, I don't get redirected, just no response. Am I configuring something incorrectly?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c7ddfb2f-5dea-5e31-2a27-02a4dc16104c
* Version Independent ID: 12ca0b9a-2638-a4ce-c45f-d4579c83dafb
* Content: [Enforce HTTPS in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-2.1&tabs=visual-studio)
* Content Source: [aspnetcore/security/enforcing-ssl.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/enforcing-ssl.md)
* Product: **asp.net-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Did you set the port ...

```csharp
services.AddHttpsRedirection(options =>
{
    options.HttpsPort = 5001;
}); 
```

... or by setting the port via one of the other approaches in the topic?

**Comment by anorborg** [comment]
Yes, indeed. I don't get a 404 either, just an empty response via curl:

```
 services.AddHttpsRedirection(options =>
            {
                options.RedirectStatusCode = StatusCodes.Status301MovedPermanently;
                options.HttpsPort = 5001;
            });
```

```
curl: (52) Empty reply from server
```
